libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2127,6 +2130,67 @@ void GCS_MAVLINK::send_raw_imu()
 #endif
 }
 
+#if AP_MAVLINK_MSG_HIGHRES_IMU_ENABLED
+void GCS_MAVLINK::send_highres_imu()
+{
+#if AP_INERTIALSENSOR_ENABLED
+    const AP_InertialSensor &ins = AP::ins();
+    const Vector3f& accel = ins.get_accel();
+    const Vector3f& gyro = ins.get_gyro();
+    uint16_t fields_updated = (HIGHRES_IMU_UPDATED_XACC | HIGHRES_IMU_UPDATED_YACC | HIGHRES_IMU_UPDATED_ZACC |
+        HIGHRES_IMU_UPDATED_XGYRO | HIGHRES_IMU_UPDATED_YGYRO | HIGHRES_IMU_UPDATED_ZGYRO);
+    uint8_t accel_id = ins.get_primary_accel();
+    
+    Vector3f mag = Vector3f();
+#if AP_COMPASS_ENABLED
+    const Compass &compass = AP::compass();
+    if (compass.get_count() >= 1) {
+        mag = compass.get_field() * 1000.0f; // convert to gauss
+    }
+    fields_updated |= (HIGHRES_IMU_UPDATED_XMAG | HIGHRES_IMU_UPDATED_YMAG | HIGHRES_IMU_UPDATED_ZMAG);
+#endif
+
+    float press_abs = 0.0f;
+    float temperature = 0.0f;
+    float altitude = 0.0f;
+    float diff_pressure = 0.0f;
+#if AP_BARO_ENABLED
+    const AP_Baro &barometer = AP::baro();
+    press_abs = barometer.get_pressure() * 0.01f;
+    temperature = barometer.get_temperature();
+    altitude = barometer.get_altitude_AMSL();
+    diff_pressure = press_abs - barometer.get_ground_pressure() * 0.01f;
+    fields_updated |= (HIGHRES_IMU_UPDATED_ABS_PRESSURE | HIGHRES_IMU_UPDATED_DIFF_PRESSURE |
+        HIGHRES_IMU_UPDATED_PRESSURE_ALT | HIGHRES_IMU_UPDATED_TEMPERATURE);
+#endif
+
```

Comment:
```suggestion

// or maybe have this as a define?
    const all_fields =  (HIGHRES_IMU_UPDATED_XACC | HIGHRES_IMU_UPDATED_YACC | HIGHRES_IMU_UPDATED_ZACC | 
        HIGHRES_IMU_UPDATED_XGYRO | HIGHRES_IMU_UPDATED_YGYRO | HIGHRES_IMU_UPDATED_ZGYRO | 
        HIGHRES_IMU_UPDATED_XMAG | HIGHRES_IMU_UPDATED_YMAG | HIGHRES_IMU_UPDATED_ZMAG |
        HIGHRES_IMU_UPDATED_ABS_PRESSURE | HIGHRES_IMU_UPDATED_DIFF_PRESSURE |
        HIGHRES_IMU_UPDATED_PRESSURE_ALT | HIGHRES_IMU_UPDATED_TEMPERATURE);

    if (fields_updated == all_fields) {
        fields_updated |= HIGHRES_IMU_UPDATED_ALL; // probably best to |= instead of = in case a GCS is checking each field.
        }
```